lib/kadm5/ipropd_slave.c

```diff
@@ -563,6 +565,9 @@ receive_everything(krb5_context context, int fd,
 
     reinit_log(context, server_context, vno);
 
+    ret = mydb->hdb_set_sync(context, mydb, 1);
+    if (ret)
+        krb5_warn(context, ret, "failed to sync the received HDB");
```

Comment:
I think that sync failure needs to be fatal.  You're clobbering `ret` with the return value of `hdb_close`.